lib/active_record/connection_adapters/cockroachdb_adapter.rb

```diff
@@ -181,7 +181,10 @@ def supports_partial_index?
       end
 
       def supports_expression_index?
-        @crdb_version >= 2122
+        # Expression indexes are partially supported by CockroachDB v21.2,
```

Comment:
if we do support it partially, then i would argue we should skip the test instead, or add a new condition for `ON CONFLICT` support and use that to skip the test / anything relevant in the ORM

otherwise, is this is a "regression" in ORM behaviour?

Reply:
Moved this to disable `supports_insert_conflict_target?` instead.

Reply:
we didn't need the `public_send` stuff right?

Reply:
It looks like it is already there for the ones I overrode.
```
%w[
  supports_savepoints?
  supports_partial_index?
  supports_partitioned_indexes?
  supports_expression_index?
  supports_insert_returning?
  supports_insert_on_duplicate_skip?
  supports_insert_on_duplicate_update?
  supports_insert_conflict_target?
  supports_optimizer_hints?
  supports_datetime_with_precision?
].each do |method_name|
  define_method method_name do
    ActiveRecord::Base.connection.public_send(method_name)
  end
end
```

Reply:
does this mean it will also disable `ON CONFLICT column_name`?

Reply:
Yeah it looks like there is no way to say we support expression indexes and `ON CONFLICT column_name`, but not `ON CONFLICT expression` without modifying activerecord itself.